tests/execute.rs

```diff
@@ -118,38 +177,126 @@ fn execute_script_generic() {
-    new_test_ext().execute_with(|| {
-        assert_eq!(1, 0);
-    });
-}
+    pub const ALICE_ADDR: &str = "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY";
+
+    let (bob_addr_32, bob_addr_mv) = addrs_from_ss58(BOB_ADDR).unwrap();
+    let (alice_addr_32, alice_addr_mv) = addrs_from_ss58(ALICE_ADDR).unwrap();
 
-#[test]
-#[ignore = "to be implemented"]
-/// Test execution of a script with correct parameters but as a wrong user.
-/// Transaction does not require sudo but call was signed with sudo.
-fn execute_script_wrong_user() {
     new_test_ext().execute_with(|| {
-        assert_eq!(1, 0);
+        // Bob publishes the move-module 'Counter', test preparation.
+        let module = assets::read_module_from_project("get-resource", "Counter");
+        assert_ok!(MoveModule::publish_module(
+            RuntimeOrigin::signed(bob_addr_32.clone()),
+            module,
+            MAX_GAS_AMOUNT,
+        ));
+
+        // Check, that there are no counters available.
+        assert_eq!(
+            get_vm_resource(&bob_addr_mv, "Counter", "Counter", &alice_addr_32).unwrap(),
+            None
+        );
+        assert_eq!(
+            get_vm_resource(&bob_addr_mv, "Counter", "Counter", &bob_addr_32).unwrap(),
+            None
+        );
+
+        // Alice and Bob execute a script to create a counter by using move-module 'Counter'.
+        let transaction_bc = script_transaction("get-resource", "create_counter", &alice_addr_mv);
+        assert_ok!(MoveModule::execute(
+            RuntimeOrigin::signed(alice_addr_32.clone()),
+            transaction_bc,
+            MAX_GAS_AMOUNT,
+        ));
+
+        let transaction_bc = script_transaction("get-resource", "create_counter", &bob_addr_mv);
+        assert_ok!(MoveModule::execute(
+            RuntimeOrigin::signed(bob_addr_32.clone()),
+            transaction_bc,
+            MAX_GAS_AMOUNT,
+        ));
+
+        // Verify counter has been created.
+        let counter = get_vm_resource(&bob_addr_mv, "Counter", "Counter", &alice_addr_32)
+            .unwrap()
+            .expect("Couldn't find Alice's counter");
+        assert_eq!(counter, vec![0, 0, 0, 0, 0, 0, 0, 0]);
+
+        let counter = get_vm_resource(&bob_addr_mv, "Counter", "Counter", &bob_addr_32)
+            .unwrap()
+            .expect("Couldn't find Bob's counter");
+        assert_eq!(counter, vec![0, 0, 0, 0, 0, 0, 0, 0]);
+
+        // Execute script that counts that created counter, but only for Alice.
+        let transaction_bc = script_transaction("get-resource", "count", &alice_addr_mv);
+        assert_ok!(MoveModule::execute(
+            RuntimeOrigin::signed(alice_addr_32.clone()),
+            transaction_bc,
+            MAX_GAS_AMOUNT,
+        ));
+
+        // Verify counter has been increased by 1.
+        let counter = get_vm_resource(&bob_addr_mv, "Counter", "Counter", &alice_addr_32)
+            .unwrap()
+            .expect("Could not find Alice's counter");
```

Comment:
```suggestion
            .expect("Couldn't find Alice's counter");
```

To be in sync with the similar message below about Bob's counter :smile: